specifyweb/frontend/js_src/lib/components/Header/Notifications.tsx

```diff
@@ -109,7 +112,7 @@ export function Notifications({
           timeout =
             document.visibilityState === 'hidden'
               ? undefined
-              : globalThis.setTimeout(doFetch, pullInterval);
+              : globalThis.setTimeout(() => doFetch(new Date()), pullInterval);
```

Comment:
This is wrong behavior. When fetching again, it should only fetch notifications that have been created since the last time we fetched. Thus, you should store the date it was last fetched in some variable and use that next time you fetch, rather than `new Date()`

Reply:
Also, you need to update the code to append newly fetched notifications to the notifications state, rather than reply all existing notifications with newly fetched